Refactored events reference documentation

mercul3s opened the pull request:
Added examples and event specification.

mercul3s committed:
Refactored events reference documentation

Added examples and event specification.

palourde commented on content/sensu-core/2.0/reference/events.md:
```diff
@@ -1,65 +1,192 @@
-
+## How do Events work?
+
+A Sensu Event is created every time a check result is processed by the Sensu server,
```
I don't think `Event` should be capitalized here and below!

palourde commented on content/sensu-core/2.0/reference/events.md:
```diff
@@ -1,65 +1,192 @@
+### Attributes
+timestamp    | 
+-------------|------ 
+description  | The time of the Event occurrence in eopoch time. 
```
s/eopoch/epoch/

palourde commented on content/sensu-core/2.0/reference/events.md:
```diff
@@ -1,65 +1,192 @@
+  }
+{{</ highlight >}}
+
+## Examples
```
I feel like these examples don't belong here, since they use `sensuctl`, and instead we should simply put a full JSON event. That being said, maybe we could create a guide around events? But I'm not sure what it could consist of, except maybe resolving events?

mercul3s replied:
OK, I'll convert these to JSON. I don't think we need a full guide for events as I think the only thing of note is resolving/viewing events, and that's pretty easy to figure out from the cli help text.

mercul3s committed:
Addressing PR feedback

Removed cli examples in favor of example json event. Fixed typo and
lowercased event throughout the doc.

mercul3s commented:
@palourde Fixed the typos and added a JSON example event in place of sensuctl examples.

palourde commented on content/sensu-core/2.0/reference/events.md:
```diff
@@ -1,65 +1,254 @@
+regardless of the status indicated by the check result. An event is created by 
+the agent on receipt of the check execution result. The agent will execute any configured
+[hooks][4] the check might have. From there, it is forwarded to the Sensu backend 
+for processing. Potentially noteworthy Events may be processed by one or more 
```
s/Events/events/

mercul3s replied:
whoops I thought I got them all 😅

palourde commented on content/sensu-core/2.0/reference/events.md:
```diff
@@ -1,65 +1,254 @@
+    ],
+    "subscriptions": [
+      "web",
+      "entity:example-entity"
```
It's really a small detail but for consistency, if the entity ID is `example-agent`, the subscription should be `entity:example-agent`.

mercul3s replied:
D'oh, I intended for them to be the same! Fixed.

palourde reviewed (approved):
Awesome work!!!

![giphy](https://user-images.githubusercontent.com/4105580/37984267-d582ba2c-31c3-11e8-951a-c61c3cd5f7bd.gif)
